lua/nvim-training/utility.lua

```diff
@@ -7,25 +7,51 @@ function utility.set_buffer_to_lorem_ipsum_and_place_cursor_randomly()
 	utility.update_buffer_respecting_header(lorem_ipsum)
 	utility.move_cursor_to_random_point()
 end
+
 function utility.create_highlight(x, y, len)
 	vim.api.nvim_set_hl(0, "UnderScore", { underline = true })
 	vim.api.nvim_buf_add_highlight(0, internal_config.global_hl_namespace, "UnderScore", x, y, y + len)
+	return string.sub(utility.get_line(x), y+1, y+len)
 end
 
 function utility.move_cursor_to_random_point()
 	vim.api.nvim_win_set_cursor(0, utility.calculate_random_point_in_text_bounds())
 end
 
+function utility.highlight_random_word()
+	local x = utility.random_line_index()
+	local line = utility.get_line(x)
+	local wordParams = utility.get_word_params(line)
+	local selected = wordParams[math.random(1, #wordParams)]
+	return utility.create_highlight(x, selected[1]-1, selected[2])
```

Comment:
This function should not have a return value, especially not something unrelated to its tasks. Maybe just remove it? But then you have to rework the yank task accordingly.